R/messages.R

```diff
@@ -110,3 +110,64 @@ setvar <- function(core, dates, var, values, unit)
 
     invisible(NULL)
 }
+
+
+#' Fetch all the available Hector variables
+#'
+#' This function is similar to fetchvars, except it fetches all available
+#' Hector variables, including exogenous inputs (e.g., emissions & pre-industrial
+#' concentrations), parameters (e.g., alpha, beta, ECS), and output
+#' variables (e.g., concentrations & forcings).
+#'
+#' @param core Hector core object
+#' @param dates Vector of dates, optional
+#' @param scenario Scenario name, optional str
+#' @param outfile Absolute path, including the name, of the output .csv file.
+#' @return vars_all    Dataframe containing all Hector variables
+#' @family main user interface functions
+#' @export
+fetchvars_all <- function(core, dates=NULL, scenario=NULL, outfile=NULL)
+{
+
+    # Get output for variables that do use the date param
+    rslt_date <- fetchvars(core, dates, vars_date, scenario)
+
+    # Get variables that DO NOT use date arg
+    rslt_nodate <- fetchvars(core, NA, vars_nodate, scenario)
+
+    rslt <- rbind(rslt_date, rslt_nodate)
+
+    if (!is.null(outfile)) {
+        # Check that outfile ends with the .csv file extension
+        if (substr(outfile, nchar(outfile)-3, nchar(outfile)) != ".csv") {
+            stop("'outfile' must end in '.csv'")
+        }
+
+        # Pivot the dataframe "wide" so years run along the x axis
+        # Catch warning from reshape()
+        suppressWarnings(rslt_pretty <- reshape(rslt, direction="wide",
+                                        idvar=c("scenario", "variable", "units"),
+                                        timevar="year"))
+
+        # Clean up the column names (value.YYYY --> YYYY)
+        col_names <- colnames(rslt_pretty)[-1:-3]
```

Comment:
This sequence of transformations is complex and hard to follow, and I think there's an easier way to do it.  First, it isn't necessary to chop off the first three columns to rename the remaining ones.  Also, `sub` can operate on a vector, so you should be able to just write:
```R
colnames(rslt_pretty) <- sub('value\\.', "", colnames(rslt_pretty))
```

After that, if you want to rearrange the columns in the data frame, you can just access the data frame as if it were a list
```R
nc <- ncol(rslt_pretty)
rslt_pretty <- rslt_pretty[c(1,2,3, nc, seq(4, nc-1))]
```
And I think those three lines together give you what you want (though, I confess, I haven't tested it).